### PR TITLE
Build basic capability

### PR DESCRIPTION
**Pull request by forevermatt**
This implements the tests and code for syncing all users and syncing a specific user, minus the actual calls to the Insite ID Store. I plan to add the sync-users-changed-sync-x in a separate feature branch.

### CODE REVIEW

File: application/frontend/config/main.php [hunk 1]
```diff
@@ -1,32 +1,43 @@
 <?php
 
+use Sil\Idp\IdSync\common\components\ApiConsumer;
```
Comment (wcjr) [inline application/frontend/config/main.php]: I noticed much more namespacing, wondering if I need to do the same on the broker?

File: application/frontend/config/main.php [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
     // http://www.yiiframework.com/doc-2.0/guide-structure-applications.html#controllerNamespace
-    'controllerNamespace' => 'frontend\controllers',
+    'controllerNamespace' => 'Sil\\Idp\\IdSync\\frontend\\controllers',
+    
     'components' => [
+        
         // http://www.yiiframework.com/doc-2.0/guide-security-authentication.html
         'user' => [
-            'identityClass' => 'common\models\ApiConsumer', // custom Bearer <token> implementation
+            'identityClass' => ApiConsumer::class, // custom Bearer <token> implementation
```
Comment (wcjr) [inline application/frontend/config/main.php]: much better, I'll remember to change that in the broker as well.

**Comment by wcjr** [comment]
lgtm, go for it!

File: application/frontend/config/main.php (hunk 1 above)
Comment (forevermatt) [inline application/frontend/config/main.php]: I think we're tending towards a namespace of `Sil\Idp\_RepoNamespace_\...` for code related to this IdP work, so it probably wouldn't hurt. It's probably not worth a ton of time to go back and refactor stuff, though, if any of it doesn't cooperate.

Comment (wcjr) [inline application/frontend/config/main.php]: no biggie, I'll get to it later: https://github.com/silinternational/idp-id-broker/issues/40

**Pull request merged**